lib/govobject/govobject.js

```diff
@@ -164,7 +164,15 @@ GovObject.prototype.fromBufferReader = function(reader) {
     var dataHex = reader.read(reader.buf.length);
 
     var object = JSON.parse(dataHex.toString('utf8'));
-    var jsonObject = object[0][1];// attempt to parse as JSON Object
+
+    // attempt to parse as JSON Object
+    var jsonObject;
+    if (object.constructor.name === 'Object') {
+      jsonObject = object;
+    } else if (object.constructor.name === 'Array') {
+      jsonObject = object[0][1];
+    }
+
```

Comment:
Good catch. I would expect that it would not, but I would also expect the previous line to break if it weren't a nested array, so would the same result happen here?

Reply:
I've just added an error case. May not be the most elegant, but at least we can merge this.